src/JointAdapter.cpp

```diff
@@ -88,14 +87,19 @@ void JointVelocityAdapter::update(
   if (std::isnan(pidVelocity))
     throw std::range_error("calculated pidVelocity is NaN");
 
-  if (desiredVelocity + pidVelocity > mVelocityLimit)
+  auto finalVelocity = desiredVelocity + pidVelocity;
+  auto upperVelLimit = mDof->getVelocityUpperLimit();
```

Comment:
I think it would be better to read this on construction, rather than reading it every time we need to update the JointAdapter.

Reply:
@brianhou Done!